lib/extensions/ar_virtual.rb

```diff
@@ -66,43 +66,99 @@ module ClassMethods
     #
 
     def virtual_delegate(*methods)
-      options = methods.pop
-      unless options.kind_of?(Hash) && options[:to]
+      options = methods.extract_options! || {}
+      unless (to = options[:to])
         raise ArgumentError, 'Delegation needs an association. Supply an options hash with a :to key as the last argument (e.g. delegate :hello, to: :greeter).'
       end
-      delegate(*methods, options)
+
+      to = to.to_s
+      if to.include?(".") && methods.size > 1
+        raise ArgumentError, 'Delegation only supports specifying a method name when defining a single virtual method'
+      end
+
+      location = caller_locations(1, 1).first
+      file, line = location.path, location.lineno
```

Comment:
@NickLaMuro I didn't have much success with `caller_locations(2,1).first`, but I'll try again. it sure would make it easier to read and closer to delegate